### PR TITLE
feat: adding clippy and rustfmt config

### PR DESCRIPTION
**Pull request by hopinheimer**
Adding the formatting and clippy config before we start

### CODE REVIEW

File: .clippy.toml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# This feature is in Rust nightly, as of v1.62. The line is added in anticipation.
+# Once the feature stabilizes, remove the #[allow(clippy::unwrap_used)] lines in front of every `mod test`
+allow-unwrap-in-tests = true
```
Comment (KolbyML) [inline .clippy.toml:3]: add an extra line to remove 
image

File: rustfmt.toml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+reorder_imports  = true
+imports_granularity = "Crate"
+group_imports = "StdExternalCrate"
+wrap_comments = true
+comment_width = 100
+normalize_comments = true
+format_code_in_doc_comments = true
```
Comment (KolbyML) [inline rustfmt.toml:7]: same comment as above

**Review by KolbyML — APPROVED** [review]
:shipit:

**Comment by hopinheimer** [comment]
first LGTM!

**Review by syjn99 — APPROVED** [review]

**Pull request merged**